gcsstore/gcsservice.go

```diff
@@ -342,19 +343,27 @@ func (service *GCSService) FilterObjects(params GCSFilterParams) ([]string, erro
 			return nil, err
 		}
 
+		if strings.HasSuffix(objAttrs.Name, "info") {
+			continue
+		}
 		split := strings.Split(objAttrs.Name, "_")
 
+		// If the object name does not split on "_", we have a composed object.
 		// If the object name splits on "_" in to four pieces we
 		// know the object name we are working with is in the format
 		// [uid]_tmp_[recursion_lvl]_[chunk_idx]. The only time we filter
 		// these temporary objects is on a delete operation so we can just
 		// append and continue without worrying about index order
-		if len(split) == 4 {
+
+		switch len(split) {
+		case 1:
+			names = []string{objAttrs.Name}
+			break loop
```

Comment:
I am wondering if this labelled break could be replaced with a simple `continue` since this would jump to the beginning of the for loop?

Reply:
This is a break because in the case of the partitions and the composed file remaining in GCS (say you have 3 chunks and the composed file already, but the delete failed on the chunks) the iterator doesn't proceed to go through the chunks. If a composed file is found (in the case of `len(split) == 1`, we should just exit out of the loop entirely,.

Reply:
Oh my bad, I confused the labeled break with a goto statement and thought that this would jump to the beginning of the loop but instead it terminates the loop. In that case, it's all good :+1: